lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,162 @@
+
+    my %dna_samples_by_individual_id;
+    for my $dna_sample (@dna_samples) {
+        push @{$dna_samples_by_individual_id{$dna_sample->individual->id}}, $dna_sample;
+    }
+    my @individual_ids = sort keys %dna_samples_by_individual_id;
+    $self->status_message('Found '. scalar(@individual_ids) .' indivudals');
+
+    my @subject_mappings;
+    for my $individual_id (@individual_ids) {
+        my $individual = Genome::Individual->get($individual_id);
+        my @individual_dna_samples = @{$dna_samples_by_individual_id{$individual_id}};
+        my $individual_dna_sample_count = scalar(@individual_dna_samples);
+        $self->status_message('Found '. $individual_dna_sample_count .' samples for individual '. $individual->__display_name__);
+
+        if ( $individual_dna_sample_count > 2) {
+            $self->fatal_message('This command currently does not support more than 2 DNA samples per individual.');
+        }
+
+        my @normal_samples;
+        my @tumor_samples;
+
+        for my $individual_dna_sample (@individual_dna_samples) {
+            my $dna_sample_common_name = $individual_dna_sample->common_name || "NULL";
+            push(@normal_samples, $individual_dna_sample) if ($dna_sample_common_name =~ /$normal_sample_common_names/i);
+            push(@tumor_samples,  $individual_dna_sample) if ($dna_sample_common_name =~ /$tumor_sample_common_names/i);
+        }
+
+        if (scalar(@normal_samples) > 1) {
+            $self->fatal_message('More than one normal DNA sample was specified for this individual: '. $individual->__display_name__);
+        }
+        if (scalar(@tumor_samples) > 1) {
+            $self->fatal_message('More than one tumor DNA sample was specified for this individual: '. $individual->__display_name__);
+        }
+        my %data;
+        if (@normal_samples && @tumor_samples) {
+            %data = (
+                tumor_sample => $tumor_samples[0]->$id_method,
+                normal_sample => $normal_samples[0]->$id_method,
+                snv_variant_list_id => '',
+                indel_variant_list_id => '',
+                sv_variant_list_id => '',
+                tag => 'somatic',
+            );
+        } elsif (@normal_samples) {
+            %data = (
+                tumor_sample => $normal_samples[0]->$id_method,
+                normal_sample => '',
+                snv_variant_list_id => '',
+                indel_variant_list_id => '',
+                sv_variant_list_id => '',
+                tag => 'germline',
+            );
+        } elsif (@tumor_samples) {
+            %data = (
+                tumor_sample => $tumor_samples[0]->$id_method,
+                normal_sample => '',
+                snv_variant_list_id => '',
+                indel_variant_list_id => '',
+                sv_variant_list_id => '',
+                tag => 'tumor-only',
+            );
+        }
+        # TODO: See if subject mapping already exists before adding to the list
```

Comment:
Subject Mappings store their Subjects and their Tags in hang-off tables, so there's no constraint on duplicates.  I believe the downside of duplicates is that it makes CQID take longer as it tries to get or create the same model once per mapping.